models/iaf_psc_alpha.cpp

```diff
@@ -382,15 +388,18 @@ iaf_psc_alpha::handle( SpikeEvent& e )
 {
   assert( e.get_delay_steps() > 0 );
 
+  const index input_buffer_slot = kernel().event_delivery_manager.get_modulo(
+    e.get_rel_delivery_steps( kernel().simulation_manager.get_slice_origin() ) );
+
   const double s = e.get_weight() * e.get_multiplicity();
 
   if ( e.get_weight() > 0.0 )
   {
-    B_.ex_spikes_.add_value( e.get_rel_delivery_steps( kernel().simulation_manager.get_slice_origin() ), s );
+    B_.input_buffer_.add_value( input_buffer_slot, Buffers_::SYN_EX, s );
   }
   else
   {
-    B_.in_spikes_.add_value( e.get_rel_delivery_steps( kernel().simulation_manager.get_slice_origin() ), s );
+    B_.input_buffer_.add_value( input_buffer_slot, Buffers_::SYN_IN, s );
   }
 }
```

Comment:
With all the beauty that MultiRingBuffer provides, I am tempted to suggest (`s` has same sign as `e.get_weight()`)
```
B_.input_buffer( e.get_delivery_slot(), s > 0 ? Buffers_::SYN_EX : Buffers_::SYN_IN, s );
```
What do you think?

Reply:
Done!